web_client/views/imageViewerWidget/geojs.js

```diff
@@ -93,6 +93,19 @@ var GeojsImageViewerWidget = ImageViewerWidget.extend({
                 features: ['point', 'line', 'polygon']
             });
             this._layers[annotation.id] = layer;
+
+            var setBounds = _.bind(function () {
+                var zoom = this.viewer.zoom(),
+                    bounds = this.viewer.bounds(),
+                    zoomRange = this.viewer.zoomRange();
+                if (annotation.setView) {
+                    annotation.setView(bounds, zoom, zoomRange.max, _.bind(function () { this.drawAnnotation(annotation); }, this));
```

Comment:
I think this could be simplified by rerendering to the `g:fetched` event instead.